### PR TITLE
feat: optimize IconSpinner

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Optimizes the IconSpinner module
- Adds Cosmos fixtures
- Create `cosmos` constant file with shared items.

https://github.com/cowprotocol/cowswap/assets/31534717/4f7adcee-6e45-4daf-8e8d-9181480d9a15

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]

File: apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx [hunk 1]
```diff
@@ -62,24 +62,28 @@ const Wrapper = styled.div<{ size: number; spinnerWidth: number; bgColor?: strin
...
-`;
+`
 
-export function IconSpinner({ currency, image, size = 24, children, bgColor, spinnerWidth = 2 }: IconSpinnerProps) {
+export function IconSpinner({
+  currency,
+  image,
+  size = 24,
+  children,
+  bgColor = '--cow-container-bg-01',
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx:72]: Is it possible to get this value from a const or even better, make these color names an ENUM?

Imagine if we want to pass something else.
How will we know what values are expected?

Comment (fairlighteth) [reply apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx:72]: > Imagine if we want to pass something else.
How will we know what values are expected?

I think there we probably need to agree to only always pass CSS color variables? Can you elaborate how an ENUM would add value here?

Comment (alfetopito) [reply apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx:72]: The colors can be defined in a central place, while also being more legible.

For example:

```ts
enum ColorVariables = {
  CONTAINER_BG_01 = '--cow-container-bg-01'
}

type Props = {
  // ...
  bgColor: ColorVariables
}

function Component({
  // ...
  bgColor = CONTAINER_BG_01
}: Props) {
  // ...
```

Note: code is not tested, might not work or need tweaks.

Comment (fairlighteth) [reply apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx]: Updated!

**Comment by fairlighteth** [comment]
Merging to keep this going. @alfetopito I've addressed your ENUM comment. Feel free to comment on this post-merge.

File: apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx [hunk 2]
```diff
@@ -64,12 +64,16 @@ const Wrapper = styled.div<{ size: number; spinnerWidth: number; bgColor: string
   }
 `
 
+enum ColorVariables {
+  CONTAINER_BG_01 = '--cow-container-bg-01',
+}
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx:69]: Yes, that's what I meant!
Sort of 😅 

This is the idea, but the suggestion is to put all of the variable (names?) in this enum.
Then you use the enum in the source css where they are declared, and when naming them individually.

Comment (alfetopito) [reply apps/cowswap-frontend/src/common/pure/IconSpinner/index.tsx:69]: This change by itself is breaking the build.

**Pull request merged**